### DISCUSSION TITLE
Documentation Tracker

### DISCUSSION THREAD

**Discussion by 5e-Cleric** [naturalcrit/homebrewery#4370] [General]
## About the issue

This issue should keep open to keep track of all important documentation and its changes.

Creation  motivated by a collection of issues related between them such as updating of the FAQ page, landing page, readme,  and the creation of a how to contribute document.

## List of documents

- Readme
    - [ ] Numbered list update
    - [ ] How to contribute/create a fork
- Landing page
    - [ ] #2791
- Frequent Asked Questions page (FAQ)
    - [ ] #2791
- Introduction for new developers (draft)

**Comment by 5e-Cleric** [comment]
## Introduction for new developers

Back when i started here, i had lots and lots of questions that i didn't ask because i didn't want to bother everyone, specially in the devs chat, so i kept them to myself and been asking ever since, here i collect the ones i believe are most important for people starting to contribute and those who want to join the team for a long time.

- How are projects organized, what structure do they have?
- What responsibilities does a member of the team have?
- Can i talk in the public gitter chat? is it okay if i go in and ask things?
- I'm not a software engineer or a developer, can i contribute? can i be a member of the team?
- I've seen the mods/devs around for a long time, but some answer some questions and not others, are they specialized?
- is the team language friendly?(meaning do they speak other languages apart from english?
- How can i become a member of the team?
- Who's the boss? is there a boss? how does the hierarchy work? how do you keep organized?
- What tools are at my disposal as a dev?

will update if i think of something else
This is a draft, feel free to suggest changes please

**Comment by calculuschild** [comment]
Answering this without worrying too hard about the grammar or brevity for now:

- How are projects organized, what structure do they have?
  -  *There are two projects under the Naturalcrit team. Homebrewery (the D&D Markdown app), and "Naturalcrit", which currently just acts as a login system. At some point, the orignal author (stolksdorf) had planned Naturalcrit to host a larger set of apps using a common login, but really the Homebrewery has taken over as the core site.*
- What responsibilities does a member of the team have?
  -  *There are a handful of things you can do to contribute to the project, whatever you feel comfortable with:*
    1) Write code Pull Requests to add new features or fix bugs. Clean/refactor old code for legibility or fix typos.
    2) Help manage issues. Go through the existing issues and clean out duplicates or old ones that are already solved. Help answer questions that don't require new code to fix ("I lost my brew, what do I do?"). Find issues that have been stuck or abandoned that need to be closed for good or brought back into focus.
    3) Answer questions on Reddit. We get lots of questions. Make notes of common ones so we can add them to a FAQ, or create issues on Github for issues reported on Reddit so we can track a solution.
    4) Create new CSS themes for homebrews. We can help you add it to the code if you "aren't a coding person".
    5) Review existing PRs.
    6) Give feedback on issues or on the Gitter chat. Are you a UI/UX expert/ graphic designer/ RPG enthusiast who doesn't know how to code but wants to share your input?
- Can i talk in the public gitter chat? is it okay if i go in and ask things?
  - *Yeah, it's fine to talk there. It's mostly intended for use by the devs to discuss active issues in real time, but feel free to ask questions to say hi. In some cases we might refer you to the subreddit because some answers might benefit more from the larger community (general questions about CSS, help making themes, etc.)*
- I'm not a software engineer or a developer, can i contribute? can i be a member of the team?
  - *Yep! See "what responsibilities does a team member have" above.*
- I've seen the mods/devs around for a long time, but some answer some questions and not others, are they specialized?
  - *There is a bit of specialization, but we are all just volunteers who stumbled onto the project at different times. See "Meet the Team".*
- is the team language friendly?(meaning do they speak other languages apart from english?
  - *We speak a few languages, but can also try to accommodate with translation tools if needed. See "Meet the Team".*
- How can i become a member of the team?
  - *Anyone can contribute code or provide feedback or just chat on Reddit or the gitter without any formal "becoming part of the team" process. If you hang around enough and we see that you are likely to stay for a while, and have been contributing (code/thoughts/etc.), we can grant you more access to the repo to directly edit/delete/close issues, etc., or make you a mod on the subreddit. But don't feel that having that access makes you any more or less a team member. Most of the "core team" have just been around for a while.*
- Who's the boss? is there a boss? how does the hierarchy work? how do you keep organized?
  - *See "Meet the Team".*
- What tools are at my disposal as a dev?
  - *You can install a copy of the repository on your personal computer following the Readme file. From there, you can get a code editor (VSCode, etc.), a Git manager (Github Desktop/SourceTree, etc.) and you can start contributing code. We don't really "supply" any tools to you. Just download your preferred tools for free. (not totally sure if this answers the question).

### The team:

The "team" is always welcoming new people. Some stop by only occasionally, others contributed just once, and others stuck around for a while before moving on to other projects. But those of us you'll see most often and active on the Subreddit or Gitter (the "core" team?) include:

**calculuschild** - The owner of the repository and longest on the project. Manages the accounts for the Heroku server, MongoDB Atlas database, domain name subscription, Google API account, and Patreon. Final say in merging PRs. Not really specialized in anything, but regular contributor to all parts of the repository (JS, CSS, graphics, etc.). If there's a boss it's calculuschild, but mostly he just holds the keys. From the US. Speaks English and (non-technical) Spanish. 

**G-Ambatte** - 

**Garumoo** - 

**Gazook89** - 

**Jeddai** - 

**5e-Cleric** -

**Comment by Gazook89** [comment]
> There are two projects under the Naturalcrit team. 

It should be noted that there are individual repos for each (with links), and that the vast majority of work is done on the Homebrewery repo.

> What responsibilities does a member of the team have? 

We don't have a 'code of conduct' but responsibilities include being respectful of other devs and users, answering questions positively, and contributing with humility.  

> We speak a few languages,

Maybe add that additional languages is a positive, particularly if we go down the internationalization path for UI.

> How can i become a member of the team? 

It is likely best to start with small things just to get a sense of the flow both in the app and the contribution process.  And that more than one contributor has started from zero programming experience to become regular contributors, and are happy to help earnest beginners.

> Meet the Team

**Gazook89 - ** Spent a couple years refining CSS skills using the Homebrewery, helping users make changes to their visual themes.  Started to learn Javascript by contributing a few hotkeys and snippets, and has since made more involved contributions particularly to the UI and a few other places.  Doesn't know a thing about backend, or D&D. Active on reddit (Gazook89) and Discord of Many Things (LongJohnJones).

**Comment by 5e-Cleric** [comment]
> How are projects organized, what structure do they have?
There are two projects under the Naturalcrit team. Homebrewery (the D&D Markdown app), and "Naturalcrit", which currently just acts as a login system. At some point, the orignal author (stolksdorf) had planned Naturalcrit to host a larger set of apps using a common login, but really the Homebrewery has taken over as the core site.

I actually meant projects inside the HB, like brew themes or language UI, i didn't want to call them PRs because it is not technically the same thing, but this is another good point.

___

> Answer questions on Reddit. We get lots of questions. Make notes of common ones so we can add them to a FAQ, or create issues on Github for issues reported on Reddit so we can track a solution.

I will add the Discord as well

_____

Structured as is, it could be added as a second page in the FAQ, or in a separate doc that we send to people who ask

**5e-Cleric - ** Just a fellow homebrewer who used homebrewery so much that fell in love with CSS. Created some templates before the most famous ones came to be, and has interest in bringing them all into the tool. Specializes in Snippets and UI/UX. Easily accessible via Reddit (u/5e-Cleric), in the DOMT as "5e-Cleric | Homebrewery Dev". He checks reddit and discord every day!

**Comment by 5e-Cleric** [comment]
I'm in two minds about this, should i make it a structured doc like the landing page, or questions like the FAQ?

**Comment by Gazook89** [comment]
Might just be best to update the existing [Contributing.md](https://github.com/naturalcrit/homebrewery/blob/master/contributing.md).  I don't think there needs to be any special page on the Homebrewery itself for something aimed at developers.

**Comment by calculuschild** [comment]
> Might just be best to update the existing [Contributing.md](https://github.com/naturalcrit/homebrewery/blob/master/contributing.md).

Yes, I was assuming that was the goal here. Full agreement.

**Comment by G-Ambatte** [comment]
My apologies for the delay, for some reason I thought had I already responded to this.

---

**G-Ambatte -** Knows a little about a lot, and a lot about a little. I started BASIC programming as a teen, learned C for embedded systems at my first job, and a later job saw me migrate to Python and Node. I stumbled into D&D in late 2019 as a Forever DM and became a user of Homebrewery almost immediately.
I started out as a Homebrewery contributor by creating a script to install Homebrewery locally in a FreeBSD jail. The contribution that I'm most proud of (so far) is the brew movement functions (match brew preview to editor/match editor to brew preview).
Usually available on Reddit (/u/Gambatte) and the Discord of Many Things (Gambatte | Homebrewery Loon). Speaks English, Jackspeak, and a disturbing amount of High Valyrian.

**Comment by dbolack-ab** [comment]
Should we transition this from a Topic to a Discussion? Issues really shouldn't be immortal...

**Comment by 5e-Cleric** [comment]
Not the worst idea, also not very important, transitioning it now.
